### DISCUSSION TITLE
How to by pass Content Security Policy on Firefox

### DISCUSSION THREAD

**Discussion by newptcai** [Tampermonkey/tampermonkey#2049] [Q&A]
Hi,

I have noted that in the Firefox addson Tampermonkey, there is not a section called Security on the Setting page of Tampermonkey. I wonder is there a way to bypass CSP on Firefox without turning it off completely?

I want to inject MathJax on `https://chatgpt.com`. But the CSP blocks it from running.

**Comment by derjanb** [comment]
You have to set  `Config mode` to `Advanced` first.

**Comment by newptcai** [comment]
I still get

```
Content-Security-Policy: The page’s settings blocked a script (script-src-elem) at https://cdn.mathjax.org/mathjax/latest/MathJax.js from being executed because it violates the following directive: “script-src-elem 'unsafe-inline' https://auth0.openai.com/ https://challenges.cloudflare.com/ https://*.chatgpt.com https://*.oaistatic.com https://apis.google.com/ https://chat.openai.com/ https://chatgpt.com/ https://docs.google.com/ https://jidori.g1.internal.services.openai.org/ https://js.live.net/v7.2/OneDrive.js https://oaistatic.com/ https://snc.apps.openai.com/ https://tcr9i.chat.openai.com/ https://widget.intercom.io/ https://www-onepick-opensocial.googleusercontent.com/ https://js.intercomcdn.com/ https://js.stripe.com/ wss://*.chatgpt.com”
```

I have turned on "Modifying existing content security policy headers"

![Screenshot from 2024-05-08 10-09-42](https://github.com/Tampermonkey/tampermonkey/assets/982197/0ec8b7a6-3b01-4911-838c-48b8470371d3)

**Comment by newptcai** [comment]
I see. I have to set CSP to "Remove Entirely" for this to work.
